RequestsPlus/RequestsPlus.py

```diff
@@ -0,0 +1,150 @@
+        self._name = self.__class__.__name__
+        self.apiBaseURL = apiBaseURL
+        self.contentType = contentType
+        self.authZToken = authZToken
+        self.authZType = authZType
+        self.session = requests.Session()
+        self.session.headers.update(self._prepareHeaders())
+
+    @staticmethod
+    def responseCollection(response):
+        """
+        Convenience method to make a ResponseCollection
+        object for a response.
+
+        :param response: requests Response object, usually the first of multiple pages
+        :type response: requests.models.Response
+        :return: ResponseCollection object containing multiple response pages
+        :rtype: ResponseCollection
+        """
+        return ResponseCollection(response)
+
+    @property
+    def _authZHeader(self):
+        return {} if self.authZToken is None else \
+            {HTTP_HEADER_AUTHORIZATION: ' '.join([self.authZType, self.authZToken + ''])}
+
+    @property
+    def _contentTypeHeader(self):
+        return {} if self.contentType is None else \
+            {HTTP_HEADER_CONTENT_TYPE: self.contentType}
+
+    def _prepareHeaders(self):
+        headers = {}
+        headers.update(self._contentTypeHeader)
+        headers.update(self._authZHeader)
+        return headers
+
+    def _prepareURL(self, apiQueryURI):
+        """
+        If the URI (actually just a partial URL, usually the path part) doesn't begin with
+        the base URL for the API, concatenate the two into a new URL and return it.
+
+        :param apiQueryURI: URI (actually, just a partial URL, usually the path part) for an API entry point.
+        :type apiQueryURI: str
+        :return: URL for the API query, ready for use
+        :rtype: str
+        """
+        assert isinstance(apiQueryURI, str)
+        assert not util.stringContainsAllCharacters(apiQueryURI, '{}'), \
+            'apiQueryURI contains unformatted arguments: "%s"' % apiQueryURI
+
+        if apiQueryURI.startswith(self.apiBaseURL):
+            return apiQueryURI
+
+        return urlnorm.norm(self.apiBaseURL + '/' + apiQueryURI)
+
+    def _sendRequest(self, httpMethod, apiQueryURI, **kwargs):
+        """
+        Append the specified query URI to the base URL,
+        which is then sent to the REST API using the specified method.
+
+        :param httpMethod: Name of the HTTP method used for the query
+        :type httpMethod: str
+        :param apiQueryURI: URI for the query, to be appended to the base URL
+        :type apiQueryURI: str
+        :return: Response object
+        :rtype: requests.Response
+        """
+        preparedAPIQueryURL = self._prepareURL(apiQueryURI)
+        response = None
+        sessionRequestMethod = self.session.__getattribute__(httpMethod)
+
+        try:
+            response = sessionRequestMethod(preparedAPIQueryURL, **kwargs)
+        except requests.exceptions.RequestException as e:
+            print self._name + ' error: ' + e.message
```

Comment:
In reference to the last comment I made, printing out something like this may prove useful.